### PR TITLE
EDU-920 batch room invitations

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-920

### CODE REVIEW

File: src/components/room-invitation-creation-modal.js [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+import inputValidators from '../utils/input-validators.js';
 import RoomApiClient from '../api-clients/room-api-client.js';
 import { useSessionAwareApiClient } from '../ui/api-helper.js';
 
+const FormItem = Form.Item;
+const { TextArea } = Input;
+
 const logger = new Logger(import.meta.url);
 
+const normalizeEmails = value => value.toLowerCase();
```
Comment (adrianaluchian) [inline src/components/room-invitation-creation-modal.js:16]: This should be `normalizeEmail`, as it deals with just one at a time.

Comment (adrianaluchian) [reply src/components/room-invitation-creation-modal.js:16]: No, sorry, got it, it's applied over the input, not the extracted email 😬

File: src/components/room-invitation-creation-modal.js [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+const extractEmails = value => value.match(/[^\s;,]+/g) || [];
```
Comment (adrianaluchian) [inline src/components/room-invitation-creation-modal.js:17]: How about `const extractEmails = value => (value.match(/[^\s;,]+/g) || []).filter(item => item);`
just to take out possible altogether missing emails (if ever that is the case)?
I'm thinking that those don't fall under the "invalid email" category 🤔

File: src/components/room-invitation-creation-modal.js [hunk 3]
```diff
@@ -46,16 +77,29 @@ function RoomInvitationCreationModal({ isVisible, onOk, onCancel, roomId }) {
...
       visible={isVisible}
       okButtonProps={{ loading: isLoading }}
       forceRender
       >
-      <Form form={form} name="new-room-invitation-form" initialValues={initialFormValues} onFinish={handleFormFinish} layout="vertical">
-        <EmailFormItem name="email" />
+      <Form
+        form={form}
+        layout="vertical"
+        name="send-room-invitations-form"
```
Comment (adrianaluchian) [inline src/components/room-invitation-creation-modal.js:90]: Tiny thing: does the form need to have `name` set?

File: src/components/room-invitation-creation-modal.js (hunk 2 above)
Comment (ahelmberger) [reply src/components/room-invitation-creation-modal.js:17]: I had something similar first, then changed the regex. This regex prevents any empty string matches, so `filter(item => item)` would never take out any item.

**Comment by adrianaluchian** [comment]
Maybe we should put an info icon next to the input label, telling the user what separators they can use between email addresses.

File: src/components/room-invitation-creation-modal.js (hunk 2 above)
Comment (adrianaluchian) [reply src/components/room-invitation-creation-modal.js:17]: Ah I see, smart regex, smart.

File: src/components/room-invitation-creation-modal.js (hunk 3 above)
Comment (ahelmberger) [reply src/components/room-invitation-creation-modal.js]: I assumed as it was there before, but I just tried and I don't think it's necessary, so I removed it.

**Comment by ahelmberger** [comment]
> Maybe we should put an info icon next to the input label, telling the user what separators they can use between email addresses.

Nice idea! Done.

**Comment by adrianaluchian** [comment]
> > Maybe we should put an info icon next to the input label, telling the user what separators they can use between email addresses.
> 
> Nice idea! Done.

👍 Cool
Should we in this case move it to the end of the label? Looks rather weird with the required mark.

Screen Shot 2022-11-10 at 17 25 37

Or maybe we can remove the requirement altogether and just not do anything when the user presses "Send" without any emails written?

**Comment by ahelmberger** [comment]
> > > Maybe we should put an info icon next to the input label, telling the user what separators they can use between email addresses.
> > 
> > 
> > Nice idea! Done.
> 
> +1 Cool Should we in this case move it to the end of the label? Looks rather weird with the required mark.
> Or maybe we can remove the requirement altogether and just not do anything when the user presses "Send" without any emails written?

I actually did both as it looks best. Validation will kick in anyways when no email is specified.

**Comment by adrianaluchian** [comment]
> > > > Maybe we should put an info icon next to the input label, telling the user what separators they can use between email addresses.
> > > 
> > > 
> > > Nice idea! Done.
> > 
> > 
> > +1 Cool Should we in this case move it to the end of the label? Looks rather weird with the required mark.
> > Or maybe we can remove the requirement altogether and just not do anything when the user presses "Send" without any emails written?
> 
> I actually did both as it looks best. Validation will kick in anyways when no email is specified.

Wunderbar!

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**